### DISCUSSION TITLE
Smooth scroll issue in WPF

### DISCUSSION THREAD

**Discussion by karlovm** [cefsharp/CefSharp#4138] [General]
Scrolling in WPF isn't animated. The scroll bar just steps. WinForms works correctly.

MinimalExample: the same problem

[Watch video](https://www.youtube.com/watch?v=GdkajT7CxjU)

**Comment by amaitland** [comment]
Converted to discussion as [Bug Report](https://github.com/cefsharp/CefSharp/blob/master/.github/ISSUE_TEMPLATE/bug_report.md#bug-report) was not completed.

Smooth scrolling is not implemented in the `WPF` version as each frame is rendered as a bitmap. We'd have to somehow mimic the behaviour. Related discussion https://magpcss.org/ceforum/viewtopic.php?f=6&t=18027 and for background read https://github.com/cefsharp/CefSharp/wiki/General-Usage#offscreen-rendering-osr


You can switch to using https://github.com/cefsharp/CefSharp.Wpf.HwndHost which hosts the native implementation in a `HwndHost`. Airspace issues apply.
